ambari-server/src/main/java/org/apache/ambari/server/api/query/render/ClusterBlueprintRenderer.java

```diff
@@ -106,17 +116,28 @@
       resultTree.addChild(new HashSet<>(), configType);
     }
 
+    if (resultTree.getChild(Resource.Type.ClusterSetting.name()) == null) {
+      resultTree.addChild(new HashSet<>(), Resource.Type.ClusterSetting.name());
+    }
+
+    String serviceGroupType = Resource.Type.ServiceGroup.name();
+    if (resultTree.getChild(serviceGroupType) == null) {
+      resultTree.addChild(new HashSet<>(), serviceGroupType);
+    }
```

Comment:
Can you please extract and reuse this logic?

Reply:
fixed